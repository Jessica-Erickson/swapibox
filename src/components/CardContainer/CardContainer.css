.CardContainer {
  grid-row: 2 / 3;
  grid-column: 2 / 3;

  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  overflow: scroll;
}

.default,
.favorites-default {
  color: gold;
  text-align: center;
  font-size: 2rem;
  margin-bottom: 0;
}

.favorites-default-text {
  color: gold;
  font-size: 1.75rem;
  margin-top: 0;
  align-self: flex-start;
  text-align: center;
}

.display-none {
  display: none;
}

.people-loading {
  background-image: url(../../assets/images/skywalker-fett.gif);
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
}

.planets-loading {
  background-image: url(../../assets/images/hoth.gif);
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
}

.vehicles-loading {
  background-image: url(../../assets/images/landspeeder.gif);
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
}