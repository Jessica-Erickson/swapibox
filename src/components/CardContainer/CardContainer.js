import React from 'react';
import * as API from '../../helper.js';
import PropTypes from 'prop-types';
import Card from '../Card';
import './CardContainer.css';

const CardContainer = ({ favorites , category , addFavorite , removeFavorite }) => {
  let contents;

  if (favorites.length === 0 && category === 'favorites') {
    return (
      <div>
        <h2 className="favorites-default">
          You currently don't have any favorites.
        </h2>
        <p className="favorites-default-text">
          Click on the lightsaber in the top right corner of a card to favorite it.
        </p>
      </div>
    )
  } else if (category === 'favorites') {
    contents = favorites;
    return makeCards();
  } else if (category === 'people') {
    getContents();
    return makeCards();
  } else if (category === 'planets') {
    getContents();
    return makeCards();
  } else if (category === 'vehicles') {
    getContents();
    return makeCards();
  }

  const getContents = async () => {
    if (!localStorage.getItem(category)) {
      contents = JSON.parse(localStorage.getItem(category));
    } else {
      contents = await API[category]();
    }
  }

  const makeCards = () => {
    if (typeof contents === 'array') {
      return contents.map((item) => {
        return <Card
                  contents={item}
                  addFavorite={() => addFavorite(item)}
                  removeFavorite={() => removeFavorite(item.id)}
                  isActive={favorites.includes(item)}
                  key={item.name} />
      });
    } else {
      return <h2>Data not found. Please try again later.</h2>
    }
  }
}


CardContainer.propTypes = {
  favorites: PropTypes.array.isRequired,
  category: PropTypes.string.isRequired,
  addFavorite: PropTypes.func.isRequired,
  removeFavorite: PropTypes.func.isRequired
}

export default CardContainer;