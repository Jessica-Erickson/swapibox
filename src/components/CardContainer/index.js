import React from 'react';
import './CardContainer.css';
import PropTypes from 'prop-types';
import Card from './../Card';

const CardContainer = ({ cardContents }) => {
  let cards;

  if ( cardContents[0] === '') {
    cards = <h1 className="default">Select a Category!</h1>;
  } else {
    cards = cardContents.map(thingy => {
      return <Card contents={thingy} />
    })
  }

  return (
    <main className="CardContainer">
      {
        cards
      }
    </main>
  );
}

CardContainer.propTypes = {
  cardContents: PropTypes.array.isRequired
}

export default CardContainer;
