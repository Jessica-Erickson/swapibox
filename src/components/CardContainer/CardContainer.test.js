import React from 'react'
import { shallow } from 'enzyme'
import CardContainer from './index'

describe('CardContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CardContainer cardContents={[]} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
