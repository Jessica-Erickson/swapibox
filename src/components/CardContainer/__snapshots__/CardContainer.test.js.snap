// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
<main
  className="CardContainer"
>
  <h1
    className="default"
  >
    Select a Category!
  </h1>
</main>
`;

exports[`CardContainer should render cards when it gets contents 1`] = `
<CardContainer
  cardContents={
    Array [
      Object {
        "name": "Luke Skywalker",
      },
      Object {
        "name": "Leia Organa",
      },
      Object {
        "name": "R2-D2",
      },
    ]
  }
>
  <main
    className="CardContainer"
  >
    <Card
      contents={
        Object {
          "name": "Luke Skywalker",
        }
      }
      key="Luke Skywalker"
    >
      <article
        className="Card isAlsoOff"
      >
        <h3
          className="name isWhite"
        >
          Luke Skywalker
        </h3>
        <div
          className="saber isOff"
        />
        <img
          alt="The hilt of a lightsaber. Turn the lightsaber on to favorite this card."
          className="hilt"
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          src="lightsaber-wt.png"
        />
        <ul>
          <li>
            Homeworld: 
          </li>
          <li>
            Species: 
          </li>
          <li>
            Population: 
          </li>
        </ul>
      </article>
    </Card>
    <Card
      contents={
        Object {
          "name": "Leia Organa",
        }
      }
      key="Leia Organa"
    >
      <article
        className="Card isAlsoOff"
      >
        <h3
          className="name isWhite"
        >
          Leia Organa
        </h3>
        <div
          className="saber isOff"
        />
        <img
          alt="The hilt of a lightsaber. Turn the lightsaber on to favorite this card."
          className="hilt"
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          src="lightsaber-wt.png"
        />
        <ul>
          <li>
            Homeworld: 
          </li>
          <li>
            Species: 
          </li>
          <li>
            Population: 
          </li>
        </ul>
      </article>
    </Card>
    <Card
      contents={
        Object {
          "name": "R2-D2",
        }
      }
      key="R2-D2"
    >
      <article
        className="Card isAlsoOff"
      >
        <h3
          className="name isWhite"
        >
          R2-D2
        </h3>
        <div
          className="saber isOff"
        />
        <img
          alt="The hilt of a lightsaber. Turn the lightsaber on to favorite this card."
          className="hilt"
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          src="lightsaber-wt.png"
        />
        <ul>
          <li>
            Homeworld: 
          </li>
          <li>
            Species: 
          </li>
          <li>
            Population: 
          </li>
        </ul>
      </article>
    </Card>
  </main>
</CardContainer>
`;
